python/setup.py

```diff
@@ -28,8 +28,10 @@ def getReadMe():
         "Programming Language :: Python :: 3 :: Only",
     ],
     include_package_data=True,
-    data_files=[("ml4ir/build", ["ml4ir/build/Dockerfile", "ml4ir/build/run_driver.sh"])],
     license="ASL 2.0",
     python_requires=">=3.7",
     install_requires=load_dependencies(),
+    extras_require={
```

Comment:
@jakemannix This is the key change like we discussed.

Reply:
Instead of `pip install ml4ir[spark]`, I can also make it `pip install ml4ir[full]` to be more easily understandable.

Reply:
I wonder if `[all]` is better than `[full]`?  As in "ml4ir-with-all-dependencies"?  What do other python projects do?

Reply:
The examples I saw were using `[full]`. Although, they were on stackoverflow.

Reply:
I have renamed this to `all` as I saw some more examples.
Ref -> https://hanxiao.io/2019/11/07/A-Better-Practice-for-Managing-extras-require-Dependencies-in-Python/